backend/module/audit/audit.go

```diff
@@ -52,35 +51,71 @@ func (m *mod) Register(r module.Registrar) error {
 }
 
 func (m *mod) GetEvents(ctx context.Context, req *auditv1.GetEventsRequest) (*auditv1.GetEventsResponse, error) {
+	resp := &auditv1.GetEventsResponse{}
+
+	var start time.Time
+	var end *time.Time
+
 	switch req.GetWindow().(type) {
 	case *auditv1.GetEventsRequest_Range:
 		timerange := req.GetRange()
-		start := timerange.StartTime.AsTime()
-		end := timerange.EndTime.AsTime()
-
-		events, err := m.client.ReadEvents(ctx, start, &end)
-		if err != nil {
-			return nil, err
-		}
-		return &auditv1.GetEventsResponse{Events: events}, nil
+		start = timerange.StartTime.AsTime()
+		endTime := timerange.EndTime.AsTime()
+		end = &endTime
 	case *auditv1.GetEventsRequest_Since:
 		if err := req.GetSince().CheckValid(); err != nil {
 			return nil, fmt.Errorf("problem parsing duration: %w", err)
 		}
 
 		window := req.GetSince().AsDuration()
-		start := time.Now().Add(-window)
-
-		events, err := m.client.ReadEvents(ctx, start, nil)
-		if err != nil {
-			return nil, err
-		}
-		return &auditv1.GetEventsResponse{Events: events}, nil
+		start = time.Now().Add(-window)
 	default:
 		return nil, errors.New("no time window requested")
 	}
+
+	var page int
+	if req.PageToken != "" {
+		p, err := strconv.Atoi(req.PageToken)
+		if err != nil || p < 0 {
+			return nil, fmt.Errorf("invalid page token: %s", req.PageToken)
+		}
+		page = p
+	}
+
+	limit := int(req.Limit)
+	if req.Limit == 0 {
+		limit = 1
```

Comment:
is this the proper default?

Reply:
This is only used when determining the page offset if the request limit is 0. I left a response to your comment below about the limit but this behavior maintains backwards compatibility with todays behavior where if you request audit events all are returned.